### ISSUE TITLE
Entity body response approach

### ISSUE THREAD

**Issue by cameronfletcher** [LykkeCorp/guidelines#4]
This issue was raised off the back of a demonstration of an external API.

When returning an entity body in the response for a resource located at a URL eg. `cats/fluffy` where `fluffy` is the cat's name (I know this isn't a good example):
```json
{
  "url": ".../cats/fluffy",
  "name": "fluffy",
  "age": 3,
  "owner": {
    "url": ".../owners/bob",
    "name": "bob",
    "number_of_cats": 3
  }
}
```

- Is there any reason why we should _not_  be returning a summary of the owner `bob` within the full response for the cat `fluffy`?
- And, is there some way that we can craft this into a guildeline?
eg. you are encouraged to return a summary of linked resources where it makes sense in the context of data a typical consumer of your API might have to otherwise make a subsequent request to retrieve.

For completeness, a public example of an API taking this approach is the GitHub API.
eg. https://api.github.com/orgs/dddlib/repos returns a summary of owner and https://api.github.com/users/dddlib contains the full owner details.

**Comment by jchannon** [comment]
Yes this is called resource expansion. By default you won't want to do this. If your API support is and the client request more embedded information the API may include extra information like the owner.

**Comment by JoeStead** [comment]
If we are going down this route, I'd rather use a defined format, rather than creating yet another one.

Siren is an example: https://github.com/kevinswiber/siren
Collection+Json is another one: http://amundsen.com/media-types/collection/examples/

There's quite a few out there with established communities around them driving the standardisation of these things. It'd be my preference to use a (subset) of a publicly defined one
